src/pint/models/pulsar_binary.py

```diff
@@ -163,7 +185,42 @@ def setup(self):
         self.update_binary_object(None)
 
     def validate(self):
-        super(PulsarBinary, self).validate()
+        super().validate()
+        if (
+            hasattr(self, "SINI")
+            and self.SINI.value is not None
+            and not 0 <= self.SINI.value <= 1
+        ):
+            raise ValueError(
+                f"Sine of inclination angle must be between zero and one ({self.SINI.quantity})"
+            )
+        if hasattr(self, "M2") and self.M2.value is not None and self.M2.value < 0:
+            raise ValueError(
+                f"Companion mass M2 cannot be negative ({self.M2.quantity})"
+            )
+        if (
+            hasattr(self, "ECC")
+            and self.ECC.value is not None
+            and not 0 <= self.ECC.value <= 1
+        ):
+            raise ValueError(
+                f"Eccentricity ECC must be between zero and one ({self.ECC.quantity})"
+            )
+        if self.A1.value is not None and self.A1.value < 0:
+            raise ValueError(
+                f"Projected semi-major axis A1 cannot be negative ({self.A1.quantity})"
+            )
+        if self.PB.value is not None:
+            if self.PB.value <= 0:
+                raise ValueError(
+                    f"Binary period PB must be non-negative ({self.PB.quantity})"
+                )
+            if self.FB0.value is not None:
+                raise ValueError("Model cannot have values for both FB0 and PB")
+        if self.FB0.value is not None and self.FB0.value <= 0:
```

Comment:
Should we have a check that Both `FB` and `PB` are None?

Reply:
I think that's checked by the function immediately below this, `check_required_params`.